src/cliques.c

```diff
@@ -0,0 +1,628 @@
+  if (include_obj != Fail) {
+    set_bit_array_from_gap_list(CLIQUE, include_obj);
+    complement_bit_arrays(get_conditions(TRY, 0), CLIQUE, nr);
+    for (uint16_t i = 1; i <= LEN_LIST(include_obj); ++i) {
+      intersect_bit_arrays(get_conditions(TRY, 0),
+                           GRAPH->neighbours[INT_INTOBJ(ELM_LIST(include_obj, i)) - 1],
+                           nr);
+    }
+  }
+  // Update TRY using exclude_obj
+  if (exclude_obj != Fail) {
+    BitArray* exclude = new_bit_array(MAXVERTS); 
+    set_bit_array_from_gap_list(exclude, exclude_obj);
+    complement_bit_arrays(get_conditions(TRY, 0), exclude, nr);
+  }
+  
+  // Get the isolated vertices of the graph
+  BitArray* isolated = new_bit_array(MAXVERTS);
+  Int first_isolated = -1;
+  for (uint16_t i = 0; i < nr; ++i) {
+    if (size_bit_array(GRAPH->neighbours[i], nr) == 0) {
+      if (first_isolated == -1 && get_bit_array(get_conditions(TRY, 0), i)) {
+        first_isolated = i;
+      }
+      set_bit_array(isolated, i, true);
+    }
+  }
+  // Update TRY using isolated, only one isolated vertex is used
+  if (first_isolated != -1) { 
+    complement_bit_arrays(get_conditions(TRY, 0), isolated, nr);
+    set_bit_array(get_conditions(TRY, 0), first_isolated, true);
+  }
+
+  // Discard the generators of aut_grp_obj which act on the isolated vertices
+  if (size_bit_array(isolated, nr) > 0) {
+    Obj new_group = Fail; 
+    Obj gens = CALL_1ARGS(GeneratorsOfGroup, *group);
+    DIGRAPHS_ASSERT(IS_LIST(gens));
+    for (Int i = 1; i <= LEN_LIST(gens); ++i) {
+      Obj s = CALL_1ARGS(SmallestMovedPointPerm, ELM_LIST(gens, i));
+      if (s != Infinity && !get_bit_array(isolated, INT_INTOBJ(s) - 1)) { 
+        if (new_group == Fail) {
+          new_group = CALL_1ARGS(Group, ELM_LIST(gens, i));
+        } else { 
+          new_group = CALL_2ARGS(ClosureGroup, new_group, ELM_LIST(gens, i));
+        }
+      }
+    }
+    *group = new_group;
+  }
+ 
+  if (hook_obj != Fail) {
+    GAP_FUNC = hook_obj;
+    HOOK     = clique_hook_gap;
+  } else {
+    HOOK = clique_hook_collect;
+  }
+  USER_PARAM = user_param_obj;
+ 
+  return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Main functions
+////////////////////////////////////////////////////////////////////////////////
+
+static void BronKerbosch(uint16_t  depth, 
+                         uint16_t  rep_depth,
+                         uint64_t  limit,
+                         uint64_t* nr_found,
+                         bool      max,
+                         uint16_t  size,
+                         Obj       group) { 
+
+  uint16_t nr = GRAPH->nr_vertices;
+  BitArray* try = get_conditions(TRY, 0);
+  BitArray* ban = get_conditions(BAN, 0);
+
+  if (depth > 0 && !max && ( size == 0 || size == depth)) {
+    // We are not looking for maximal cliques
+    HOOK(USER_PARAM, CLIQUE, nr);
+    *nr_found += 1;
+    if (*nr_found >= limit) {
+      longjmp(OUTOFHERE, 1);
```

Comment:
longjmp & setjmp are tricky beasts. I'd recommend to avoid them if possible. And here it really is possible, and quite easy: just change `BronKerbosch` to return an int; e.g. `0` for success, `1` for failure (or the other way around, doesn't matter). Then instead of doing a longjmp here and below, just `return FAILURE`; change all other `return;` to `return SUCCESS;`. And finally, in the recursive calls, just do `if (FAILURE == BronKerbosch(...)) return FAILURE;`